### ISSUE TITLE
Parenthesised LHS

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12486]
At the moment, all these are valid in espree:

```js
(a) = b;

(a = b) = c; // !?

[(a)] = b;

[...(a)] = b;

[(a) = b] = c;

({ a: (b) } = c);

({ a: (b) = c } = d);
```

Is anything in this list a valid syntax?

`no-extra-parens` currently doesn't report anything of the above. I'm not sure which of these should be reported in `no-extra-parens` and which should be reported to acorn ( `(a = b) = c` looks like an obvious parser bug).

**Comment by mdjermanovic** [comment]
An interesting thing with `(a = b) = c` is that both Acorn and Babel parse `a = b` here as an `AssignmentPattern`.

Babel transpiles `(a = b) = c` as a destructuring assignment, where `b` is a default value:

```js
var _c = c;
var _c2 = _c;
a = _c2 === void 0 ? b : _c2;
_c;
```

In actual Chrome `(a = b) = c` is an `Invalid left-hand side in assignment` syntax error.

**Comment by mdjermanovic** [comment]
This was actually already reported babel/babel#8796

Filed acornjs/acorn#886 now.
